src/tui/mod.rs

```diff
@@ -122,10 +122,18 @@ impl Tui {
         self.scroll_offset = min(self.scroll_max(), scroll_offset);
     }
 
+    fn maybe_follow(&mut self) {
+        let height = self.size.height as usize;
+        if self.scroll_offset >= self.line_count - height - 1 {
```

Comment:
What if `height` is greater than `line_count` (e.g. when first starting)? Might need some saturating arithmetic or some other check here.